metadata-ingestion/src/datahub/ingestion/source/data_lake.py

```diff
@@ -0,0 +1,991 @@
+import dataclasses
+import logging
+import os
+from dataclasses import field as dataclass_field
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, Iterable, List, Optional
+
+import pydantic
+import pydeequ
+from pydeequ.analyzers import (
+    AnalysisRunBuilder,
+    AnalysisRunner,
+    AnalyzerContext,
+    ApproxCountDistinct,
+    ApproxQuantile,
+    ApproxQuantiles,
+    Histogram,
+    Maximum,
+    Mean,
+    Minimum,
+    StandardDeviation,
+)
+from pyspark.conf import SparkConf
+from pyspark.sql import SparkSession
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.functions import col, count, isnan, when
+from pyspark.sql.types import ArrayType, BinaryType, BooleanType, ByteType
+from pyspark.sql.types import DataType as SparkDataType
+from pyspark.sql.types import (
+    DateType,
+    DecimalType,
+    DoubleType,
+    FloatType,
+    IntegerType,
+    LongType,
+    MapType,
+    NullType,
+    ShortType,
+    StringType,
+    StructField,
+    StructType,
+    TimestampType,
+)
+
+from datahub.configuration.common import AllowDenyPattern, ConfigModel
+from datahub.emitter.mce_builder import DEFAULT_ENV, get_sys_time
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.ingestion.source.aws.aws_common import AwsSourceConfig, make_s3_urn
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    BooleanTypeClass,
+    BytesTypeClass,
+    DateTypeClass,
+    NullTypeClass,
+    NumberTypeClass,
+    RecordTypeClass,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringTypeClass,
+    TimeTypeClass,
+)
+from datahub.metadata.schema_classes import (
+    ChangeTypeClass,
+    DatasetFieldProfileClass,
+    DatasetProfileClass,
+    DatasetPropertiesClass,
+    HistogramClass,
+    MapTypeClass,
+    OtherSchemaClass,
+    QuantileClass,
+    ValueFrequencyClass,
+)
+
+# hide annoying debug errors from py4j
+logging.getLogger("py4j").setLevel(logging.ERROR)
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+NUM_SAMPLE_ROWS = 3
```

Comment:
More of an fyi: SQL profiler samples 20 rows

Not sure we need to exactly match that behavior, but might be worth bumping this up a bit?